### PR TITLE
sys/sysmacros.h: support sysmacros header

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
Add sys/sysmacros.h 

Refer to:
https://fossies.org/linux/glibc/misc/sys/sysmacros.h
https://github.com/bminor/musl/blob/master/include/sys/sysmacros.h

Signed-off-by: Jiuzhu Dong <dongjiuzhu1@xiaomi.com>
## Impact
N/A
## Testing
Vela CI

### CODE REVIEW

File: include/sys/sysmacros.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ ****************************************************************************/
+
+#define major(x) \
+        ((unsigned)((((x) >> 31 >> 1) & 0xfffff000) | \
+                    (((x) >> 8) & 0x00000fff)))
+#define minor(x) \
+        ((unsigned)((((x) >> 12) & 0xffffff00) | ((x) & 0x000000ff)))
+
+#define makedev(x,y) ( \
+        (((x) & 0xfffff000ull) << 32) | \
```
Comment (pkarashchenko) [inline include/sys/sysmacros.h:35]: Definitely depends on `HAVE_LONG_LONG`

Comment (Donny9) [reply include/sys/sysmacros.h]: ULL to big, refer to :https://github.com/freebsd/freebsd-src/blob/833a452e9f082a7982a31c21f0da437dbbe0a39d/contrib/libarchive/libarchive/archive_pack_dev.c#L90 to change to uint32.

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/sys/sysmacros.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef __INCLUDE_SYS_SYSMACROS_H
+#define __INCLUDE_SYS_SYSMACROS_H
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define major(x)          ((int)(0x00ff & ((x) >> 8)))
+#define minor(x)          ((int)(0xffff00ff & (x)))
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply include/sys/sysmacros.h:29]: my guess is that the initial dev_t is 16bits(high 8bits for major and low 8bit for minor) and later extend to 32bits to support more devices in the same class, but the more minor bits have to put to high 16bits to keep the binary combability.

[the remark this answers is not in the document]
Comment (pkarashchenko) [reply include/sys/sysmacros.h:29]: Maybe concatenating 16 + 8 bits into continuous value can be a solution?

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply include/sys/sysmacros.h:29]: Yes, could be. I think nobody consider/require NuttX kernel need be binary compatibility.

[the remark this answers is not in the document]
Comment (pkarashchenko) [reply include/sys/sysmacros.h:29]: As this is a direct copy from FreeBSD then let's use it as is.

**Comment by fjpanag** [comment]
This PR breaks NuttX...

I am using sim, with `CONFIG_HOST_FS` enabled.  
I am trying to get the size of a file in the host FS using `stat()`.

When `dev_t` is `uint32_t` the stat struct contains garbage.  
When I revert it back to `uint16_t` it works correctly.

**Comment by Donny9** [comment]
> This PR breaks NuttX...
> 
> I am using sim, with `CONFIG_HOST_FS` enabled. I am trying to get the size of a file in the host FS using `stat()`.
> 
> When `dev_t` is `uint32_t` the stat struct contains garbage. When I revert it back to `uint16_t` it works correctly.

Okay, i'l see it right away

**Comment by xiaoxiang781216** [comment]
@fjpanag could you try where https://github.com/apache/incubator-nuttx/pull/6686 can fix your issue?

**Comment by fjpanag** [comment]
> @fjpanag could you try where #6686 can fix your issue?

Yes, indeed it solves the issue.  
Thank you!

**Pull request merged**